docs/02-app/01-building-your-application/07-configuring/15-content-security-policy.mdx

```diff
@@ -242,6 +242,27 @@ module.exports = {
 }
 ```
 
+## Built-in XSS protection
+
+Next.js provides built-in XSS protection by warning about navigations using untrusted `javascript:` URLs.
+For example, when you render a `Link` with user input, it could result in `<Link href="javascript:alert('Oh, no!')" />`.
+When that happens, you will see a warning from React in the console similar to "A future version of React will block javascript: URLs as a security precaution."".
+In addition, Next.js will also warn when such URLs are passed to `router.push(href)`, `router.replace(href)` and `router.prefetch(href)` with "A future version of Next.js will block `javascript:` URLs as a security precaution.".
+
+If you trust these URLs, you can silence this warning by using `router.push({ unsafeHref: { __href: href } })` instead.
+
+In a future version Next.js will block `javascript:` URLs by default. You can opt-in to this behavior by setting the experimental config `hardenedXSSProtection` to `true` in your `next.config.js`:
+
+```js filename="next.config.js"
+module.exports = {
+  experimental: {
```

Comment:
In the past we've used `future` for flags that will be enabled in a future major.
```suggestion
  future: {
```

Reply:
Done: 6513b89676eb96b4bb0b565a95b03e4bb774e52e